src/storage-types/src/sinks.rs

```diff
@@ -44,6 +44,63 @@ pub struct StorageSinkDesc<S: StorageSinkDescFillState, T = mz_repr::Timestamp>
     pub from_storage_metadata: <S as StorageSinkDescFillState>::StorageMetadata,
 }
 
+impl<S: Debug + StorageSinkDescFillState + PartialEq, T: Debug + PartialEq + PartialOrder>
+    StorageSinkDesc<S, T>
+{
+    /// Determines if `self` is compatible with another `StorageSinkDesc`, in
+    /// such a way that it is possible to turn `self` into `other` through a
+    /// valid series of transformations.
+    ///
+    /// Currently, the only "valid transformation" is the passage of time such
+    /// that the sink's as ofs may differ. However, this will change once we
+    /// support `ALTER CONNECTION` or `ALTER SINK`.
+    pub fn alter_compatible(
```

Comment:
If I am reading this right, this method is just validating compatibility, not altering anything yet (which it might do later)?

Reply:
Yes––this is meant to be used as a preemptive check or an assert that the transformations only occur in expected ways.